libs/data-mapper/src/lib/models/Schema.ts

```diff
@@ -64,7 +64,7 @@ export interface SchemaNodeExtended extends SchemaNode {
   pathToRoot: PathItem[];
   parentKey: string | undefined;
   // Used only in ArrayItem arrays, not regular arrays
-  arrayItemIndex: number | undefined;
+  arrayItemIndex?: number | undefined;
```

Comment:
Why make this optional?

Reply:
it was giving me errors when I was adding the mocks that I pulled directly from the store. wouldn't it only apply to certain nodes anyways?